css/src/components/progress-bar.scss

```diff
@@ -0,0 +1,116 @@
+$progress-bar-background-color: $secondary-background-glow-high-contrast !default;
+$progress-bar-value-background-color: $primary !default;
+$progress-bar-value-background-color-warning: $warning !default;
+$progress-bar-value-background-color-danger: $danger !default;
+$progress-bar-value-background-color-success: $success !default;
+$progress-bar-height: $spacer-2 !default;
+$progress-bar-height-sm: $spacer-1 !default;
+$progress-bar-border-radius: $border-radius-rounded !default;
+$progress-bar-indeterminate-animation-duration: 2s !default;
+
+.progress-bar {
+	display: block;
+	width: 100%;
+	max-width: 100%;
+	height: $progress-bar-height;
+	padding: 0;
+	border: none;
+	border-radius: $progress-bar-border-radius;
+	background-color: $progress-bar-background-color;
+	appearance: none;
+
+	&::-webkit-progress-bar {
+		background-color: $progress-bar-background-color;
+	}
+
+	&::-webkit-progress-value,
+	&::-moz-progress-bar {
+		background-color: $progress-bar-value-background-color;
+	}
+
+	// Size
+
+	&.progress-bar-sm {
+		height: $progress-bar-height-sm;
+	}
+
+	// Colors
+
+	&.progress-bar-success {
+		&::-webkit-progress-value,
+		&::-moz-progress-bar {
+			background-color: $progress-bar-value-background-color-success;
+		}
+	}
+
+	&.progress-bar-danger {
+		&::-webkit-progress-value,
+		&::-moz-progress-bar {
+			background-color: $progress-bar-value-background-color-danger;
+		}
+	}
+
+	&.progress-bar-warning {
+		&::-webkit-progress-value,
+		&::-moz-progress-bar {
+			background-color: $progress-bar-value-background-color-warning;
+		}
+	}
+
+	&:indeterminate {
+		animation-name: slide-background;
+		animation-timing-function: linear;
+		background-color: $progress-bar-background-color;
+		background-image: linear-gradient(
+			to right,
+			$progress-bar-background-color 0%,
+			$progress-bar-value-background-color 20%,
+			$progress-bar-background-color 40%
+		);
+		background-repeat: no-repeat;
+		background-position: top left;
+		background-size: 150% 150%;
+		animation-duration: $progress-bar-indeterminate-animation-duration;
+		animation-iteration-count: infinite;
+
+		[dir='rtl'] & {
```

Comment:
Can we use `:dir` here?

```suggestion
		&:dir(rtl) {
```

`[dir='rtl'] &` is susceptible to this possible issue:
```html
<div dir="rtl">
    <div dir="ltr">
        <!-- will receive RTL styles, even though the nearest ancestor is LTR -->
        <progress class="progress-bar" />
```

Reply:
Hmm great catch. How do you feel about this update:

```scss
html[dir='rtl'] &, // or body[dir='rtl'] ?
&:dir(rtl) {
  ...
}
```

I'm not sure how to treat it better. On one hand I want to have something universal, and I feel like following the `html`'s attribute should work. But at the same time there is a chance to miss overwriting rules somewhere in the body. And at the same time, I don't want to assign direction at all times to progress to make it rtl.

Reply:
never mind, i thought :dir works differently. Updated